src/components/AttachmentModal.js

```diff
@@ -128,16 +132,70 @@ class AttachmentModal extends PureComponent {
      * Close the confirm modal.
      */
     closeConfirmModal() {
-        this.setState({isConfirmModalOpen: false});
+        this.setState({isAttachmentInvalid: false});
     }
 
     /**
-     * Check if the attachment size is less than the API size limit.
      * @param {Object} file
      * @returns {Boolean}
      */
-    isValidSize(file) {
-        return !file || lodashGet(file, 'size', 0) < CONST.API_MAX_ATTACHMENT_SIZE;
+    isValidFile(file) {
+        if (file.size > CONST.API_ATTACHMENT_VALIDATIONS.MAX_SIZE) {
+            this.setState({
+                isAttachmentInvalid: true,
+                attachmentInvalidReasonTitle: this.props.translate('attachmentPicker.attachmentTooLarge'),
+                attachmentInvalidReason: this.props.translate('attachmentPicker.sizeExceeded'),
+            });
+            return false;
+        }
+
+        if (file.size < CONST.API_ATTACHMENT_VALIDATIONS.MIN_SIZE) {
+            this.setState({
+                isAttachmentInvalid: true,
+                attachmentInvalidReasonTitle: this.props.translate('attachmentPicker.attachmentTooSmall'),
+                attachmentInvalidReason: this.props.translate('attachmentPicker.sizeNotMet'),
+            });
+            return false;
+        }
+
+        const {fileExtension} = this.splitExtensionFromFileName(file.name);
+        if (!underscore.contains(CONST.API_ATTACHMENT_VALIDATIONS.ALLOWED_EXTENSIONS, fileExtension)) {
+            const invalidReasion = `${this.props.translate('attachmentPicker.notAllowedExtension')} ${CONST.API_ATTACHMENT_VALIDATIONS.ALLOWED_EXTENSIONS.join(', ')}`;
+            this.setState({
+                isAttachmentInvalid: true,
+                attachmentInvalidReasonTitle: this.props.translate('attachmentPicker.wrongFileType'),
+                attachmentInvalidReason: invalidReasion,
+            });
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * @param {Object} file
+     */
+    validateAndDisplayFileToUpload(file) {
+        if (!file) {
+            return;
+        }
+
+        if (!this.isValidFile(file)) {
+            return;
+        }
+
+        if (file instanceof File) {
```

Comment:
Kind of curious whether the `File` is going to have a `size` property. Might be good to test on native and log out the value just to be sure (can help with this if you're running into issues).